src/main/java/model/PreferencesModel.java

```diff
@@ -1,8 +1,112 @@
 package model;
 
+import java.io.*;
+
+import observe.Notifier;
 import observe.Observable;
 import observe.PreferencesObserver;
 
+import javax.xml.bind.JAXBContext;
+import javax.xml.bind.JAXBException;
+import javax.xml.bind.Marshaller;
+import javax.xml.bind.Unmarshaller;
+import javax.xml.bind.annotation.XmlElement;
+import javax.xml.bind.annotation.XmlRootElement;
+
+@XmlRootElement
 public class PreferencesModel extends Observable<PreferencesObserver> {
 
+    private boolean musicEnabled  = true;
+    private boolean soundsEnabled = true;
+
+    private final String PATH = "preferences.xml";
+
+    /**
+     * This variable has to prevent the algorithm to enter an infite loop as during the loading process
+     * a temporal PreferencesModel-instance will be created which would itself call the loading process again.
+     * Which means, the first PreferencesModel sets this variable to 'true' in order to prevent the other Models
+     * from loading the file again.
+     */
+    private static boolean objectcreating = false;
+    
+    public PreferencesModel() {
+
+        if (!objectcreating) {
```

Comment:
...While the PreferencesModel was loading its data from the file, this boolean is true. Because the Marshaller created a temporary instance of the PreferencesModel during the loading process, the constructor of the PreferencesModel was called by him, too. And this boolean prevented the temporary PreferencesModel-instance from trying to load its data from the file which would have caused an infinite loop.
However, with the new Preferences class, this boolean doesn't exist any more, as the preferences are loaded externally from the model.